src/Symfony/Component/Form/Extension/Core/Type/ColorType.php

```diff
@@ -12,9 +12,68 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 class ColorType extends AbstractType
 {
+    /**
+     * @see https://www.w3.org/TR/html52/sec-forms.html#color-state-typecolor
+     */
+    private const HTML5_PATTERN = '/^#[0-9a-f]{6}$/i';
+
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator = null)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if (!$options['html5']) {
+            return;
+        }
+
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event): void {
+            $value = $event->getData();
+            if (null === $value || '' === $value) {
+                return;
+            }
+
+            if (\is_string($value) && preg_match(self::HTML5_PATTERN, $value)) {
+                return;
+            }
+
+            $messageTemplate = 'This value is not a valid HTML5 color.';
+            $messageParameters = [
+                '{{ value }}' => is_scalar($value) ? (string) $value : \gettype($value),
```

Comment:
I suggest using `%value%` as placeholder. This follows the recommended convention of the translation component, making it compatible with the ICU format too (which automatically strips `%` chars around placeholder names, but not the `{{ ` and ` }}`).

Reply:
Alright, in this case we need to fix the FileType.

Reply:
Actually after checking, the Form component always uses `{{ value }}` (see TransformationFailureListener and FormValidator as well for example). Should we open an issue to turn everything to `%value%`?

Reply:
Actually, our `IntlFormatter` does support `{{ value }}`. Nothing to do here.